### PR TITLE
feat: api: Make ClientCalcCommP multithreaded

### PR DESCRIPTION
**Pull request by magik6k**
## Proposed Changes
Make `ClientCalcCommP` a lot faster

## Checklist

Before you mark the PR ready for review, please make sure that:
- [ ] All commits have a clear commit message.
- [ ] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

File: node/impl/client/client.go [hunk 1]
```diff
@@ -1295,16 +1278,20 @@ func (a *API) ClientCalcCommP(ctx context.Context, inpath string) (*api.CommPRet
 		return nil, xerrors.Errorf("seek to start: %w", err)
 	}
 
-	pieceReader, pieceSize := padreader.New(rdr, uint64(stat.Size()))
-	commP, err := ffiwrapper.GeneratePieceCIDFromFile(arbitraryProofType, pieceReader, pieceSize)
+	w := &writer.Writer{}
+	_, err = io.CopyBuffer(w, rdr, make([]byte, writer.CommPBuf))
+	if err != nil {
+		return nil, xerrors.Errorf("copy inte commp writer: %w", err)
```
Comment (ZenGround0) [inline node/impl/client/client.go:1284]: nit: into

File: node/impl/client/client.go [hunk 2]
```diff
@@ -1295,16 +1278,20 @@ func (a *API) ClientCalcCommP(ctx context.Context, inpath string) (*api.CommPRet
 		return nil, xerrors.Errorf("seek to start: %w", err)
 	}
 
-	pieceReader, pieceSize := padreader.New(rdr, uint64(stat.Size()))
-	commP, err := ffiwrapper.GeneratePieceCIDFromFile(arbitraryProofType, pieceReader, pieceSize)
```
Comment (ZenGround0) [inline node/impl/client/client.go:1299]: This kind of makes sense but I don't understand why this ffiwrapper thing exists and was called in the first place if the commp writer thing could just do a direct sum over the piece bytes in the first place.

Comment (magik6k) [reply node/impl/client/client.go:1299]: The writer thing is relatively recent, and this API existed for quite some time before.

Comment (magik6k) [reply node/impl/client/client.go:1299]: (the writer also calls FFI, it just does that from multiple threads for smaller chunks, then creates the top tree from them)

**Review by ZenGround0 — APPROVED** [review]

**Pull request merged**